code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -8,6 +8,10 @@
 	armor_penetration = 50
 	flags = NOBLOODY
 	can_embed = 0//No embedding pls
+	var/base_reflectchance = 40

```

Comment:
This gets subtracted by the projectile's damage divided by 3, so a laser that does 30 damage would have a reflectchance of 30, and so on. The more powerful your laser the less likely it reflects.